src/main/java/com/amazon/opendistroforelasticsearch/security/configuration/OpenDistroSecurityIndexSearcherWrapper.java

```diff
@@ -122,11 +122,8 @@ protected final boolean isBlockedIndexRequest() {
     }
 
     protected final boolean isPermittedOnIndex() {
-        final TransportAddress caller = (TransportAddress) this.threadContext.getTransient(ConfigConstants.OPENDISTRO_SECURITY_REMOTE_ADDRESS);
-        final User user = (User) threadContext.getTransient(ConfigConstants.OPENDISTRO_SECURITY_USER);
-        if (caller == null || user == null) {
-            return false;
-        }
```

Comment:
Can you explain why removing this is OK ?

I don't see any checks for null in "mapRoles" method.
https://github.com/opendistro-for-elasticsearch/security/blob/master/src/main/java/com/amazon/opendistroforelasticsearch/security/privileges/PrivilegesEvaluator.java#L436

Reply:
- There are few other places where `mapRoles()` method is called without the check for the remote address to be not null.
- The `mapRoles()` call eventually translated to the call to `Set<String> map(final User user, final TransportAddress caller)` (either in `ConfigModelV6` or `ConfigModelV7`) that correctly handles cases when `caller` is `null`